libvcx/src/api_lib/utils/logger.rs

```diff
@@ -194,41 +196,43 @@ impl LibvcxDefaultLogger {
         info!("LibvcxDefaultLogger::init >>> pattern: {:?}", pattern);
 
         let pattern = pattern.or(env::var("RUST_LOG").ok());
-        if cfg!(target_os = "android") {
-            #[cfg(target_os = "android")]
-            let log_filter = match pattern.as_ref() {
-                Some(val) => match val.to_lowercase().as_ref() {
-                    "error" => Filter::default().with_min_level(log::Level::Error),
-                    "warn" => Filter::default().with_min_level(log::Level::Warn),
-                    "info" => Filter::default().with_min_level(log::Level::Info),
-                    "debug" => Filter::default().with_min_level(log::Level::Debug),
-                    "trace" => Filter::default().with_min_level(log::Level::Trace),
-                    _ => Filter::default().with_min_level(log::Level::Error),
-                },
-                None => Filter::default().with_min_level(log::Level::Error),
-            };
-
-            //Set logging to off when deploying production android app.
-            #[cfg(target_os = "android")]
-            android_logger::init_once(log_filter);
-            info!("Logging for Android");
-        } else {
-            let formatter = match env::var("RUST_LOG_FORMATTER") {
-                Ok(val) => match val.as_str() {
-                    "text_no_color" => text_no_color_format,
+        cfg_if! {
```

Comment:
Not too sure if this cleanup is worth adding a dependency

Reply:
There is a clear separation of code for android platform and all others located in one place.

The previous version has 4 blocks guarded by `#[cfg(target_os = "android")]`.

Crate cfg_if is widely used (we already use it indirectly) and is quite stable.

Reply:
OK, convinced👌 :)